src/App.jsx

```diff
@@ -1,13 +1,117 @@
-
+import React from 'react';
 import './App.css'
-
+import FruitCounter from './components/Fruitcounter/FruitCounter.jsx';
+import Randvoorwaarden from './components/Form/Form.jsx';
 function App() {
 
-  return (
-    <>
-        <h1>Fruitmand bezorgservice</h1>
-    </>
-  )
+
+// initiele staat
+// voor elk type fruit, met beginwaarde 0
+// en waarde kunnen worden aangepast met set-functies
+
+    const [bananen, setBananen] = React.useState(0);
+    const [aardbeien, setAardbeien] = React.useState(0);
+    const [appels, setAppels] = React.useState(0);
+    const [kiwis, setKiwis] = React.useState(0);
+
+// functies om fruitwaarde te verhogen, verlagen en resetten
+    const verhoog = (fruitSetter, huidigeWaarde) => {
+        fruitSetter(huidigeWaarde + 1);
+    };
+
+    const verlaag = (fruitSetter, huidigeWaarde) => {
+        if (huidigeWaarde > 0) {
+            fruitSetter(huidigeWaarde - 1);
```

Comment:
Wist je dat je met de setters van useState al de huidige waarde tot je beschikking hebt? Je kan namelijk een functie meegeven (in plaats van de nieuwe waarde) aan die functie:
```suggestion
            fruitSetter(currentValue => currentValue - 1);
```
Niet alleen hoef je dan niet meer de huidige waarde mee te geven, maar het is ook een stuk veiliger. In theorie, **en in grote React projecten ook zeker in de praktijk**, kun je hebben dat de waarde die je meegeeft al een oude waarde is en de échte huidige waarde al anders is. Je kunt je voorstellen dat je dan ineens met verkeerde waardes aan het werk bent.